sdks/go/pkg/beam/io/xlang/debeziumio/debezium.go

```diff
@@ -0,0 +1,133 @@
+// the transforms in this package.
+//
+// The version of the expansion service should match the version of the Beam SDK
+// being used. For numbered releases of Beam, these expansions services are
+// released to the Maven repository as modules. For development versions of
+// Beam, it is recommended to build and run it from source using Gradle.
+//
+// Current supported SDKs including expansion service modules
+// * Java
+// 	  - Vendored Module: beam-sdks-java-io-debezium-expansion-service
+// 	  - Run via Gradle: ./gradlew :sdks:java:io:debezium:expansion-service:shadowJar
+//						 java -jar <path-to-debezium-jar> <port>
+//    - Reference Class: org.apache.beam.io.debezium.DebeziumIO
+package debeziumio
+
+import (
+	"reflect"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/typex"
+)
+
+// DriverClassName is the type for valid and supported Database connectors for Debezium IO.
+type DriverClassName string
+
+const (
+	// MySQL connector for Debezium
+	MySQL DriverClassName = "MySQL"
+	// PostgreSQL connector for Debezium
+	PostgreSQL = "PostgreSQL"
+)
+
+const readURN = "beam:transform:org.apache.beam:debezium_read:v1"
+
+// readFromDebeziumSchema is config schema that matches exactly with the Java's Debezium IO
+// for cross language payload.
+type readFromDebeziumSchema struct {
+	ConnectorClass       string
+	Username             string
+	Password             string
+	Host                 string
+	Port                 string
+	MaxNumberOfRecords   *int64
+	ConnectionProperties []string
+}
+
+type debeziumConfig struct {
+	expansionService string
+	readSchema       *readFromDebeziumSchema
+}
+
+// readOption facilitates additional parameters to debeziumio.Read() Ptransform.
+type readOption func(*debeziumConfig)
+
+// Read is an external PTransform which reads from Debezium and returns a
+// JSON string. It requires the address of an expansion service for Debezium IO.
+//
+// Example:
+// 	 username := "debezium"
+//   password := "dbz"
+//   host := "localhost"
+//   port := "5432"
+//   connectorClass := debeziumIO.POSTGRESQL
+//   maxrecords := 1
+//   debeziumio.Read(s.Scope("Read from debezium"), expansionAddr, username, password, host, port, connectorClass,
+//                   reflectx.String, debeziumio.MaxRecord(maxrecords), debeziumio.ExpansionService("localhost:9000"))
+func Read(s beam.Scope, username, password, host, port string, connectorClass DriverClassName, t reflect.Type, opts ...readOption) beam.PCollection {
+	rfds := readFromDebeziumSchema{
+		ConnectorClass: string(connectorClass),
+		Username:       username,
+		Password:       password,
+		Host:           host,
+		Port:           port,
+	}
+	dc := debeziumConfig{readSchema: &rfds}
+	for _, opt := range opts {
+		opt(&dc)
+	}
+	pl := beam.CrossLanguagePayload(rfds)
+	outT := beam.UnnamedOutput(typex.New(t))
+	out := beam.CrossLanguage(s, readURN, pl, dc.expansionService, nil, outT)
```

Comment:
Until we set a default automatic expansion service, we should have the function fail if `idc.expansionService == ""` not set after processing the options.

Reply:
Got it.